sql/core/src/test/scala/org/apache/spark/sql/execution/command/ShowCreateTableSuiteBase.scala

```diff
@@ -0,0 +1,145 @@
+ *   - V1 table catalog tests:
+ *     `org.apache.spark.sql.execution.command.v1.ShowCreateTableSuiteBase`
+ *     - V1 In-Memory catalog: `org.apache.spark.sql.execution.command.v1.ShowCreateTableSuite`
+ *     - V1 Hive External catalog:
+*        `org.apache.spark.sql.hive.execution.command.ShowCreateTableSuite`
+ */
+trait ShowCreateTableSuiteBase extends QueryTest with DDLCommandTestUtils {
+  override val command = "SHOW CREATE TABLE"
+
+  test("SPARK-33892: SHOW CREATE TABLE w/ char/varchar") {
+    val db = "ns1"
+    val table = "tbl"
+    withNamespaceAndTable(db, table) { t =>
+      sql(s"CREATE TABLE $t(v VARCHAR(3), c CHAR(5)) $defaultUsing")
+      val rest = sql(s"SHOW CREATE TABLE $t").head().getString(0)
+      assert(rest.contains("VARCHAR(3)"))
+      assert(rest.contains("CHAR(5)"))
+    }
+  }
+
+  test("DO NOT SUPPORT TEMP VIEW") {
+    val viewName = "spark_28383"
+    withTempView(viewName) {
+      sql(s"CREATE TEMPORARY VIEW $viewName AS SELECT 1 AS a")
+      val ex = intercept[AnalysisException] {
+        sql(s"SHOW CREATE TABLE $viewName")
+      }
+      assert(ex.getMessage.contains(
+        s"$viewName is a temp view. 'SHOW CREATE TABLE' expects a table or permanent view."))
+    }
+
+    withGlobalTempView(viewName) {
+      sql(s"CREATE GLOBAL TEMPORARY VIEW $viewName AS SELECT 1 AS a")
+      val globalTempViewDb = spark.sessionState.catalog.globalTempViewManager.database
+      val ex = intercept[AnalysisException] {
+        sql(s"SHOW CREATE TABLE $globalTempViewDb.$viewName")
+      }
+      assert(ex.getMessage.contains(
+        s"$globalTempViewDb.$viewName is a temp view. " +
+          "'SHOW CREATE TABLE' expects a table or permanent view."))
+    }
+  }
+
+  test("SPARK-36012: ADD NULL FLAG WHEN SHOW CREATE TABLE") {
+    val db = "ns1"
+    val table = "tbl"
+    withNamespaceAndTable(db, table) { t =>
+      sql(
+        s"""
+           |CREATE TABLE $t (
+           |  a bigint NOT NULL,
+           |  b bigint
+           |)
+           |USING ${classOf[SimpleInsertSource].getName}
+        """.stripMargin)
+      val showDDL = getShowCreateDDL(s"SHOW CREATE TABLE $t")
+      // if v2 showDDL(0) == s"CREATE TABLE $t ("
+      // if v1 showDDL(0) == s"CREATE TABLE `$db`.`$table` ("
+      assert(showDDL(1) == "`a` BIGINT NOT NULL,")
+      assert(showDDL(2) == "`b` BIGINT)")
+      assert(showDDL(3) == s"USING ${classOf[SimpleInsertSource].getName}")
+    }
+  }
+
+  test("SPARK-24911: KEEP QUOTES FOR NESTED FIELDS") {
+    val db = "ns1"
+    val table = "tbl"
+    withNamespaceAndTable(db, table) { t =>
+      sql(s"CREATE TABLE $t (`a` STRUCT<`b`: STRING>) $defaultUsing")
+      val showDDL = getShowCreateDDL(s"SHOW CREATE TABLE $t")
+      // if v2 showDDL(0) == s"CREATE TABLE $t ("
+      // if v1 showDDL(0) == s"CREATE TABLE `$db`.`$table` ("
+      assert(showDDL(1) == "`a` STRUCT<`b`: STRING>)")
+      if (catalogVersion == "Hive V1") {
```

Comment:
If there are diffs, please, put tests to the catalog specific test suites. This will help to unify command's output in the future otherwise we will need to review all base test suite and find all such `if`s